lib/ecto/adapters/sqlite3/connection.ex

```diff
@@ -985,6 +1012,12 @@ defmodule Ecto.Adapters.SQLite3.Connection do
         source: source,
         hints: hints
       } ->
+        if hints != [] do
+          raise Ecto.QueryError,
+            query: query,
+            message: "join hints are not supported by SQLite3"
+        end
+
```

Comment:
Should `Enum.map(hints, &[?\s | &1]),` be removed from the iolist below?

Reply:
good catch!